sktime/transformations/bootstrap/_tsbootstrap.py

```diff
@@ -0,0 +1,135 @@
+        "skip-inverse-transform": True,  # is inverse-transform skipped when called?
+        "univariate-only": False,  # can the transformer handle multivariate X?
+        "handles-missing-data": True,  # can estimator handle missing data?
+        "X-y-must-have-same-index": False,  # can estimator handle different X/y index?
+        "enforce_index_type": None,  # index type that needs to be enforced in X/y
+        "fit_is_empty": True,  # is fit empty and can be skipped? Yes = True
+        "transform-returns-same-time-index": False,
+        "python_dependencies": ["tsbootstrap>=0.0.3"],
+    }
+
+    def __init__(
+        self,
+        bootstrap,
+        include_actual=False,
+    ):
+        self.bootstrap = bootstrap
+        self.include_actual = include_actual
+        super().__init__()
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _transform must support all types in it
+            Data to be transformed
+        y : Series or Panel of mtype y_inner_mtype, default=None
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        transformed version of X
+        """
+        bootstrapped_samples = self.bootstrap.bootstrap(X, test_ratio=0)
+
+        bootstrapped_samples = [pd.DataFrame(sample) for sample in bootstrapped_samples]
+
+        boostrapped_df = pd.concat(
+            bootstrapped_samples,
+            keys=[f"synthetic_{i}" for i in range(len(bootstrapped_samples))],
+        )
+
+        _X = X.copy()
+        if self.include_actual:
+            _X.index = pd.MultiIndex.from_product([["actual"], range(len(X))])
+            return pd.concat([_X, boostrapped_df], axis=0)
+        else:
+            return boostrapped_df
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+        deps = cls.get_class_tag("python_dependencies")
```

Comment:
Just observing it's probably being done directly against the package name in other adapters. But this may be better, as future changes in dependencies won't affect.

Reply:
yes, we should do `_check_estimator_deps(self)`

Reply:
self is not available since get_test_params is a classmethod. However, we can instanitate a tsbootstrap for testing.. But I do not really like to do that.

Reply:
oh, then `cls` works too, as arg